src/irmin-pack/unix/ext.ml

```diff
@@ -593,7 +593,8 @@ module Maker (Config : Conf.S) = struct
                    implementation detail. This is safe since the callback
                    [f] is attached to [t.running_gc.promise], which is
                    referenced for the lifetime of a GC process. *)
-                let _ = Lwt.bind x.promise f in
+                let p = Lwt.bind x.promise f in
+                Lwt.on_success p Fun.id;
```

Comment:
@metanivek what do you think about this change? or we could instead use `on_failure` to log a message.

Reply:
I think it is unnecessary -- what is this line gaining for us? We could log a message on failure, but I don't see a reason to do so at this call site.

Reply:
you're probably right its unnecessary, I removed it

Reply:
:+1: To explain my reasoning (which is open to critique if it is not accurate!): `x.promise` (and `x.resolver`) exist just so that users can run code when a GC finalizes. The call to `Lwt.bind` attaches `f` to `x.promise` which means `f` will execute when we call `Lwt.wakeup_later x.resolver`. `Lwt.on_success p Fun.id` just adds `Fun.id` to create this promise chain: `x.promise` -> `f` -> `Fun.id`.

Reply:
Thanks for the explanation, makes sense. It was a "tentative" suggestion, I agree that adding a Fun.id is useless. I'm just not used on not waiting on lwt promises, in case something goes wrong. But in this case, `f` can add logs messages if it needs to.

Reply:
Yeah, I'll admit that it's a bit of an atypical usage of Lwt in that it is only using the promise part. Logging (and/or event reporting) is exactly how this callback is used in the Tezos integration.